unit/java_bytecode/java_bytecode_convert_method/convert_initalizers.cpp

```diff
@@ -0,0 +1,250 @@
+/*******************************************************************\
+
+ Module: Unit tests for converting abstract classes
+
+ Author: DiffBlue Limited. All rights reserved.
+
+\*******************************************************************/
+
+#include <testing-utils/catch.hpp>
+
+#include <memory>
+
+#include <util/config.h>
+#include <util/symbol_table.h>
+
+#include <java_bytecode/java_bytecode_language.h>
+
+#include <testing-utils/load_java_class.h>
+#include <testing-utils/require_type.h>
+
+struct test_datat
+{
+  symbol_tablet symbol_table;
+  std::string constructor_descriptor;
+};
+
+/// Verify that a given descriptor is marked as a constructor in the symbol
+/// table
+/// \param test_data The data to run the test on
+void require_is_constructor(const test_datat &test_data)
+{
+
+  const symbolt &constructor =
+    test_data.symbol_table.lookup_ref(test_data.constructor_descriptor);
+  THEN("The constructor should be marked as a constructor")
+  {
+    code_typet constructor_type = require_type::require_code(constructor.type);
+    REQUIRE(constructor_type.get_is_constructor());
+  }
+}
+
+/// Verify that a given descriptor is not marked as a constructor in the symbol
+/// table
+/// \param test_data The data to run the test on
+void require_is_static_initalizer(const test_datat &test_data)
+{
+  const symbolt &constructor =
+    test_data.symbol_table.lookup_ref(test_data.constructor_descriptor);
+  THEN("The constructor should be marked as a constructor")
+  {
+    code_typet constructor_type = require_type::require_code(constructor.type);
+    REQUIRE_FALSE(constructor_type.get_is_constructor());
+  }
+}
+
+SCENARIO(
+  "java_bytecode_convert_initalizers",
+  "[core][java_bytecode][java_bytecode_convert_method]")
+{
+  GIVEN("A class with some constructors")
+  {
+    const symbol_tablet symbol_table = load_java_class(
+      "ClassWithConstructors", "./java_bytecode/java_bytecode_convert_method");
+    std::string base_constructor_name = "java::ClassWithConstructors.<init>";
+    WHEN("Looking at the parameterless constructor")
+    {
+      require_is_constructor([&]() {
+        test_datat data;
+        data.constructor_descriptor = base_constructor_name + ":()V";
+        data.symbol_table = symbol_table;
+        return data;
+      }());
```

Comment:
Instead of using lambda, wouldn't it be easier to just construct the `test_datat` and then call `require_is_constructor` on it?

Reply:
The idea was to kind of provide like named parameters, though I think in this case I agree.